typespec-extension/src/code-model-builder.ts

```diff
@@ -2228,81 +2037,44 @@ export class CodeModelBuilder {
             }
           });
         }
-      } else {
-        // parentSchema could be DictionarySchema, which means the model is "additionalProperties"
-        pushDistinct(objectSchema.parents.all, parentSchema);
-      }
-    } else if (isRecordModelType(this.program, type)) {
-      // "pure" Record processed elsewhere
-
-      // "mixed" Record that have properties, treat the model as "additionalProperties"
-      /* type should have sourceModel, as
-      model Type is Record<> {
-        prop1: string
       }
-      */
-      const parentSchema = type.sourceModel
-        ? this.processSchema(type.sourceModel, this.getName(type.sourceModel))
-        : // process self as pure Record (instead of Model = Record + properties), do not use cache
-          this.processDictionarySchema(type, this.getName(type));
+      objectSchema.discriminatorValue = type.discriminatorValue;
+    }
+    if (type.additionalProperties) {
+      // model has additional property
+      const sdkDictType: SdkDictionaryType = {
+        kind: "dict",
+        keyType: {
+          kind: "string",
+          encode: "string",
+          nullable: false,
+        },
+        description: type.description,
+        nullableValues: false,
+        nullable: false,
+        valueType: type.additionalProperties,
+      };
+      const parentSchema = this.processDictionarySchemaFromSdkType(sdkDictType, "Record");
       objectSchema.parents = new Relations();
       objectSchema.parents.immediate.push(parentSchema);
       pushDistinct(objectSchema.parents.all, parentSchema);
```

Comment:
What if `objectSchema.parents` already have value here?

I think it be possible to have 
```ts
model Child extends Parent {
  ...Record<unknown>;
}
```

`Parent` should be processed from `type.baseModel`.

Reply:
good catch, updated